### PR TITLE
[4.0] Remove the deprecated code in the JHtmlBehavior class

### PR DESCRIPTION
**Pull request by laoneo**
Removes the deprecated code in the `JHtmlBehavior` class.

There are some more functions which are empty, but not deprecated. What should we do with them?

### CODE REVIEW

**Comment by wilsonge** [comment]
You're going to have to go back through this one I'm afraid. There's some places in the privacy component for example where we are using the old modal code for example

File: components/com_content/tmpl/article/default_links.php [hunk 1]
```diff
@@ -61,8 +61,6 @@
 								htmlspecialchars($label, ENT_COMPAT, 'UTF-8') . '</a>';
 							break;
 						case 3:
-							// Open in a modal window
```
Comment (wilsonge) [inline components/com_content/tmpl/article/default_links.php:64]: Can we drop the correct bootstrap modal code here please

File: libraries/src/Form/Field/ModulepositionField.php [hunk 2]
```diff
@@ -136,9 +136,6 @@ public function setup(\SimpleXMLElement $element, $value, $group = null)
 	 */
 	protected function getInput()
 	{
-		// Load the modal behavior script.
-		HTMLHelper::_('behavior.modal', 'a.modal');
```
Comment (wilsonge) [inline libraries/src/Form/Field/ModulepositionField.php:140]: and here. Looks like we were still using the old modal code here (although i'm not sure we're using this form field in core....)

**Comment by laoneo** [comment]
@wilsonge changed to bos modal with data attributes.

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**